.github/workflows/deploy-standalone-plugins.yml

```diff
@@ -1,99 +1,154 @@
+          zip -r ../dist/${{ matrix.plugin }}.zip .
+
+      - name: Generate checksum
+        working-directory: ./build/dist
+        run: |
+          mkdir -p $RUNNER_TEMP/plugin-checksums
+          find . -type f -print0 | sort -z | xargs -r0 shasum -a 256 -b | sed 's# \*\./# *#' > $RUNNER_TEMP/plugin-checksums/checksums.txt
+          shasum -a 256 -U -c $RUNNER_TEMP/plugin-checksums/checksums.txt
+          cat $RUNNER_TEMP/plugin-checksums/checksums.txt | while read sum file; do echo "$sum $file" > ${file#\*}.sha256; done
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.plugin }}
+          path: ./build/dist
+
+      - name: Start deployment
+        uses: bobheadxi/deployments@v1
+        id: wporg-deployment
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: "wp.org plugin: ${{ matrix.plugin }}"
+
+      - name: Deploy Plugin - ${{ matrix.plugin }}
         uses: 10up/action-wordpress-plugin-deploy@stable
         with:
-          dry-run: ${{ matrix.dry-run }}
+          dry-run: ${{ github.event_name == 'workflow_dispatch' && inputs.dry-run || false }}
         env:
-          SVN_PASSWORD: ${{ secrets.SVN_PASSWORD }}
+          SLUG: ${{ matrix.plugin }}
+          VERSION: ${{ env.PLUGIN_VERSION }}
           SVN_USERNAME: ${{ secrets.SVN_USERNAME }}
-          SLUG: ${{ matrix.slug }}
-          VERSION: ${{ steps.set_version.outputs.version }}
-          BUILD_DIR: ./${{ matrix.directory }}/${{ matrix.slug }}
-          ASSETS_DIR: ./${{ matrix.directory }}/${{ matrix.slug }}/.wordpress-org
+          SVN_PASSWORD: ${{ secrets.SVN_PASSWORD }}
+          BUILD_DIR: ./build/${{ matrix.plugin }}
+          ASSETS_DIR: ./plugins/${{ matrix.plugin }}/.wordpress-org
+
+      - name: Finish deployment
+        if: always()
+        uses: bobheadxi/deployments@v1
+        with:
+          step: finish
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: ${{ job.status }}
+          deployment_id: ${{ steps.wporg-deployment.outputs.deployment_id }}
+          env: "wp.org plugin: ${{ matrix.plugin }}"
+          env_url: "https://wordpress.org/plugins/${{ matrix.plugin }}/"
+
+  release-assets:
+    name: Add release assets
+    needs: [ pre-run, deploy ]
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: write
+    if: github.event_name == 'release'
+    strategy:
+      matrix:
+        plugin: ${{ fromJSON(needs.pre-run.outputs.matrix) }}
+    steps:
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ matrix.plugin }}
+          path: ./build/dist
+
+      - name: Upload release assets
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            ./build/dist/${{ matrix.plugin }}.zip
+            ./build/dist/${{ matrix.plugin }}.zip.sha256
```

Comment:
As far as I can tell, this will add the release assets to _any_ GitHub release, correct? Does that mean when we publish a new Performance Lab release and all relevant standalone plugins are updated as part of that, the Performance Lab release on GitHub will have all those standalone plugins' ZIP files attached?

I'm not saying this is necessarily a bad idea, I just want to make sure we have thought this through. In the future, we may consider having individual GitHub releases for the standalone plugins, so in such a scenario we probably wouldn't want to do this, but rather attach the ZIP file only to the corresponding release of _that_ plugin.

Reply:
> As far as I can tell, this will add the release assets to any GitHub release, correct? Does that mean when we publish a new Performance Lab release and all relevant standalone plugins are updated as part of that, the Performance Lab release on GitHub will have all those standalone plugins' ZIP files attached?

That's correct. 

> I'm not saying this is necessarily a bad idea, I just want to make sure we have thought this through. In the future, we may consider having individual GitHub releases for the standalone plugins, so in such a scenario we probably wouldn't want to do this, but rather attach the ZIP file only to the corresponding release of that plugin.

@felixarntz As long as we are releasing all plugins on release, IMO we can keep it. As mentioned in https://wordpress.slack.com/archives/C02KGN5K076/p1710262256960409, once we rethink on how we streamline the release this can be taken care of.

Reply:
Also, it would be better if we open a separate ticket for this and track all the discussions there.

Reply:
Addressed in https://github.com/WordPress/performance/pull/1033/commits/0d5451d8bcd9c4619a1ccce54bf7e6e981b5987c